count-things/javascript/analyze-vir-dir.js

```diff
@@ -0,0 +1,50 @@
+import { readFileSync } from 'fs';
+import { extname } from 'path';
+
+import { analyzeSource } from './analyze-source.js';
+import { summarizeJsFiles } from './summarize-js-files.js';
+
+export const analyzeVirDir = async (virDir = {}) => {
+  const report = {
+    name: virDir.name,
+    path: virDir.path,
+  };
+
+  let directories;
+  if (virDir.directories?.length > 0) {
+    directories = await Promise.all(virDir.directories.map(analyzeVirDir));
+  }
+
+  let files;
+  if (virDir.files?.length > 0) {
+    //  faster, but throws Error: ENFILE: file table overflow (errno: -23)
+    // report.files = await Promise.all(
+    //   virDir.files.map((file) =>
+    //     readFile(file.path, "utf-8").then((code) =>
+    //       Object.assign({}, file, analyzeSource(code))
+    //     )
+    //   )
+    // );
+
+    // slower, but doesn't overflow the file table
+    files = virDir.files.map((file) =>
+      extname(file.name) === '.js' &&
+      (!file.name.includes('.min.') || !file.name.includes('.bundle.'))
+        ? Object.assign(
+            {},
+            file,
+            analyzeSource(readFileSync(file.path, 'utf-8')),
+          )
```

Comment:
if possible, we might want to use ES6 syntax with the spread operator here, like:
```
? {...file, analyzeSource(readFileSync(file.path, 'utf-8'))}
```

Reply:
true thing.  I'd just been working with a lot of object transformations (`Object.entries`, `.fromEntries`) and some `Object.create` in another project so this syntax came to mind quicker